flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/BatchTableEnvironment.scala

```diff
@@ -192,10 +193,14 @@ class BatchTableEnvironment(
       name: String,
       f: AggregateFunction[T, ACC])
   : Unit = {
-    implicit val typeInfo: TypeInformation[T] = TypeExtractor
-      .createTypeInfo(f, classOf[AggregateFunction[T, ACC]], f.getClass, 0)
+    implicit val typeInfo: TypeInformation[T] = UserDefinedFunctionUtils
+      .getResultTypeOfAggregateFunction(f)
       .asInstanceOf[TypeInformation[T]]
 
+    implicit val accTypeInfo: TypeInformation[ACC] = UserDefinedFunctionUtils
```

Comment:
I agree with you. But this PR is only to fix Scala classes can't work in the Scala TableEnvironments. In general, users create Scala classes and naturally will use it in Scala TableEnvironments. Registering a Scala class into Java TableEnvironments is not recommended (should forbidden). This is the same with that Scala `T` classes of `Aggregate[T, _] and TableFunction[T]` works in Scala TableEnvironments.